### PR TITLE
Change documentation

### PR DESCRIPTION
**Pull request by wmontwe**
### :tophat: What is the goal?

Update documentation to reflect currently released version.

Change documentation to reflect structure that is used in other projects.

Remove outdated docs

### :unicorn: How is it being implemented?

Rewrote parts of the asciidoc readme

Bumped
* OkHttp 4.9.1 -> 4.9.3
* ThreeTenBP 1.5.1 -> 1.6.0
* ThreeTenABP 1.3.1 -> 1.4.0

This should help with flaky tests.

### :thinking: DOD Checklist

- [ ] My code follows the code style and naming conventions of this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated the documentation.
- [ ] I have updated the changelog accordingly.

### CODE REVIEW

File: DEVELOPING.adoc [hunk 1]
```diff
@@ -119,6 +119,101 @@ Add following changes:
 
 Releases are automatically created when a tag in the form of `v{major}.{minor}.{patch}` is added.
 
+For release management, we use link:https://help.github.com/articles/creating-releases/[GitHub Releases].So just create a new release, set the tag according to link:https://semver.org/[Semantic Versioning] and publish.
```
Comment (bitPogo) [inline DEVELOPING.adoc:122]: [GitHub Releases]. So

File: DEVELOPING.adoc [hunk 2]
```diff
@@ -119,6 +119,101 @@ Add following changes:
...
+To build the project documentation we use the link:https://github.com/asciidoctor/asciidoctor-gradle-plugin[Asciidoctor Gradle Plugin] that fits our Gradle build chain.
+
+Additionally, we use the link:https://github.com/asciidoctor/asciidoctor-diagram[AsciiDoctor Diagramm extension] which adds support for BlockDiag (BlockDiag, SeqDiag, ActDiag, NwDiag), Ditaa, Erd, GraphViz, Mermaid, Msc, PlantUML, Shaape, SvgBob, Syntrax, UMLet, Vega, Vega-Lite and WaveDrom.
+
+=== Documentation structure
+
+The base for the documentation is located in the `sdk-doc` module.It's main purpose is to configure Asciidoctor and define the style and distribution.
+You will find a `index.adoc` under `sdk-doc/src/docs/asciidoc` which is used to include documentation of other modules in one single book documentation.
+
+Every project module has a `{module-name}/src/docs/asciidoc` folder where all documentation files will be located.So every module will host it's own documentation.
```
Comment (bitPogo) [inline DEVELOPING.adoc:214]: located. So

File: README.adoc [hunk 3]
```diff
@@ -93,337 +84,352 @@ The SDK has the following dependencies:
...
+* link:CONTRIBUTING.adoc[Contributing] for details about how to report bugs and propose features.
+* link:DEVELOPING.adoc[Developing] for details about our development process and how to build and test the project.
 
-==== Documentation structure
-The base for the documentation is located in the `sdk-doc` module.It's main purpose is to configure Asciidoctor and define the style and distribution.
-You will find a `index.adoc` under `sdk-doc/src/docs/asciidoc` which is used to include documentation of other modules in one single book documentation.
+== Copyright and License
 
-Every project module has a `{module-name}/src/docs/asciidoc` folder where all documentation files will be located.So every module will host it's own documentation.
+Copyright (c) 2021 D4L data4life gGmbH / All rights reserved.
```
Comment (bitPogo) [inline README.adoc:433]: maybe bump to 2022

File: docs/03_sdk_models.adoc [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+A record *payload* can contain:
+
+. Anything that can be modeled by a FHIR (STU3 or R4) resource. From a single vital sign measurement, such as body temperature, to a complex document linking to or containing multiple attachments and measuring many megabytes in size.
+. Generic Data (called App Data)
+
+For these cases we have three types of _records_: Fhir3Record, Fhir4Record and DataRecord.
+
+==== FHIR restrictions
+
+FHIR payload must be of type DomainResource.
```
Comment (bitPogo) [inline docs/03_sdk_models.adoc:60]: Maybe _DomainResource_

File: docs/03_sdk_models.adoc [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    override val identifier: String,
+    override val resource: T,
+    override val meta: ModelContract.Meta,
+    override val annotations: Annotations
+) : CallContract.Record<T>, Record()
+----
+
+==== Data restrictions
+
+Resource type is limited to ByteArray and needs to be base64 encoded.
```
Comment (bitPogo) [inline docs/03_sdk_models.adoc:106]: Maybe _ByteArray_

**Review by bitPogo — COMMENTED** [review]
Awesome!

File: DEVELOPING.adoc (hunk 2 above)
Comment (wmontwe) [reply DEVELOPING.adoc:214]: done

File: README.adoc (hunk 3 above)
Comment (wmontwe) [reply README.adoc]: done

File: docs/03_sdk_models.adoc (hunk 5 above)
Comment (wmontwe) [reply docs/03_sdk_models.adoc:106]: done

File: docs/03_sdk_models.adoc (hunk 4 above)
Comment (wmontwe) [reply docs/03_sdk_models.adoc:60]: done

File: docs/05_sdk_api.adoc [hunk 6]
```diff
@@ -0,0 +1,812 @@
...
+==== Search
+
+Search for records with following filter options:
+
+* resourceType - class type of the searched resource (not available for data)
+* annotations - custom annotations that had been added to the records (optional)
+* startDate - the filtered records have a creation date after the start date
+* endDate - the filtered records have a creation date before the endDate
+* pageSize - define the result page size, how many records should be returned
+* offset - the offset of the result list
```
Comment (bitPogo) [inline docs/05_sdk_api.adoc:523]: This is not true anymore - see #179

**Review by bitPogo — COMMENTED** [review]
Found something more

File: docs/03_sdk_models.adoc [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+. Anything that can be modeled by a FHIR (STU3 or R4) resource. From a single vital sign measurement, such as body temperature, to a complex document linking to or containing multiple attachments and measuring many megabytes in size.
+. Generic Data (called App Data)
+
+For these cases we have three types of _records_: Fhir3Record, Fhir4Record and DataRecord.
+
+==== FHIR restrictions
+
+FHIR payload must be of type _DomainResource_.
+
+FHIR References are not actively resolved and stay as they are. Meaning they are not available in PHDP. In case referenced data points should be available, we recommend containing them in the FHIR resource, http://www.hl7.org/implement/standards/fhir/references.html#contained[see, window=_blank]. Otherwise, you need to maintain the FHIR references within the PHDP manually.
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply docs/03_sdk_models.adoc:62]: 👍

File: docs/03_sdk_models.adoc [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+FHIR payload that contains Attachment is limited to following file formats:
+
+- JPEG, including JPEG/JFIF
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply docs/03_sdk_models.adoc:66]: 👍

File: docs/04_sdk_auth.adoc [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+:copyright-year: 2019-2022
+:copyright-statement: (C) {copyright-year} {compname-legal}. All rights reserved.
+
+= Authentication/Authorization
+
+This section covers the authorization project features of the SDK.
+
+== Login
+
+The SDK automatically handles all authentication and user management tasks. The user login is managed by the {compname} auth app to ensure the safety of the user’s credentials. When the `login` functionality is invoked, the SDK opens a web view with the necessary pages. Or redirects in the case of a web-based app.
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply docs/04_sdk_auth.adoc:41]: 👍

File: docs/02_sdk_concepts.adoc [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+The {compname} {phdp-plat} uses the privacy by design approach, optimizing for the strictest possible privacy-preserving settings. Only you have access to the data which your application sends to the platform.
+
+IMPORTANT: Client-side encryption lays in the hands of the integrator application, so you must use a proper <<Provide local data storage, storage>>.
+
+=== Provide local data storage
+
+{compname} doesn't offer any client-side data storage solution. You can use the following:
+
+* icon:external-link[] http://realm.io[Realm] as an encrypted database
+* icon:external-link[] https://www.zetetic.net/sqlcipher/open-source/[SQLCipher] encryption for SQL databases
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply docs/02_sdk_concepts.adoc:50]: removed

File: docs/05_sdk_api.adoc [hunk 11]
```diff
@@ -0,0 +1,812 @@
...
+
+The {compname} API follows the CRUD pattern (create, read, update, and delete).
+
+Read operations are divided into a `fetch` and a full `download` of the record.
+
+- *Fetching* a record returns only the record itself without attachments.
+
+- *Downloading* a record returns the record with all its attachments.
+
+A fetched record that just contains attachments without the binary data could load the binary data by using `donloadAttachment`.
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply docs/05_sdk_api.adoc:63]: 👍

File: docs/05_sdk_api.adoc [hunk 12]
```diff
@@ -0,0 +1,812 @@
...
+        }
+    }
+)
+----
+
+==== Download
+
+Download a record with given ID with all contained attachments. Only for available for FHIR records.
+
+NOTE: Could result in large download bandwidth usage. Use with care in mobile or bandwidth limited environments.
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply docs/05_sdk_api.adoc:268]: 👍

File: docs/05_sdk_api.adoc [hunk 13]
```diff
@@ -0,0 +1,812 @@
...
+
+==== Create
+
+Create a record and return the created record with populated metadata. In case of FHIR records, attachments have been extracted and replaced by links pointing to the storage point. Use download or downloadAttachment to retrieve them afterwards.
+
+The SDK checks the following when creating a FHIR record:
+
+- *File format*: When users provide an unsupported file format for attachment content, the SDK throws an `DataRestrictionException.UnsupportedFileType` exception.
+
+- *File size*: When the `Attachment.data` is larger than 20 MB, the SDK throws a `DataRestrictionException.MaxDataSizeViolation` exception.
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply docs/05_sdk_api.adoc:73]: 👍

File: docs/05_sdk_api.adoc (hunk 6 above)
Comment (wmontwe) [reply docs/05_sdk_api.adoc:523]: the documentation update is only scoped to the currenty released version

Comment (bitPogo) [reply docs/05_sdk_api.adoc:523]: Ah you are right...GitHub still does not respect the merge order.

**Review by bitPogo — DISMISSED** [review]
LGTM

**Review by bitPogo — DISMISSED** [review]

**Review by bitPogo — APPROVED** [review]

**Pull request merged**